rust/src/cliwrap/cliutil.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::Result;
+use std::{thread, time, path};
+use std::os::unix::process::CommandExt;
+use nix::sys::statvfs;
+
+use crate::cliwrap;
+
+/// Returns true if the current process is booted via ostree.
+pub fn is_ostree_booted() -> bool {
+    path::Path::new("/run/ostree-booted").exists()
+}
+
+/// Returns true if /usr is not a read-only bind mount
+pub fn is_unlocked() -> Result<bool> {
+    Ok(!statvfs::statvfs("/usr")?.flags().contains(statvfs::FsFlags::ST_RDONLY))
+}
+
+/// Returns true if the current process is running as root.
+pub fn am_privileged() -> bool {
+    nix::unistd::getuid() == nix::unistd::Uid::from_raw(0)
+}
+
+/// Return the absolute path to the underlying wrapped binary
+fn get_real_bin(bin_name: &str) -> String {
+    format!("/{}/{}", cliwrap::CLIWRAP_DESTDIR, bin_name)
+}
+
+/// Wrapper for execv which accepts strings
+pub fn exec_real_wrapped<T: AsRef<str> + std::fmt::Display>(bin_name: T, argv: &[T]) -> Result<()> {
```

Comment:
But we are wrapping it, dropping privileges.

Reply:
Hmm, not sure I follow. We're not running this in the drop privs path, no? I see it used in the non drop-privs path, the unlocked path, and the `!ostree_booted` path so that we just directly pass through the same args without wrapping.